packages/frontend/core/src/modules/app-sidebar/views/index.tsx

```diff
@@ -128,9 +128,33 @@ export function AppSidebar({ children }: PropsWithChildren) {
     appSidebarService.setHovering(true);
   }, [appSidebarService]);
 
-  const onMouseLeave = useCallback(() => {
-    appSidebarService.setHovering(false);
-  }, [appSidebarService]);
+  const onMouseLeave = useCallback(
+    (e: React.MouseEvent) => {
+      const target = e.relatedTarget as HTMLElement;
+
+      // should not close sidebar when hovering on dialog or menu
+      if (
+        !(target instanceof HTMLElement) ||
+        resizing ||
+        target?.closest(
+          '[role="dialog"], [role="menu"], modal-transition-container'
+        )
+      ) {
+        return;
+      }
+
+      // handle uncaught popper
+      const popper = document.querySelector(
+        '[data-radix-popper-content-wrapper]'
+      );
+      if (popper) {
+        return;
```

Comment:
It seems that in this case the sidebar never closes, even if the dialog has been closed.

Reply:
I think we can try, when the sidebar is floating, we listen to the mousemove event on the window, and close it when the mouse moves a certain distance out of the sidebar range